prow/scripts/cluster-integration/helpers/delete-image.sh

```diff
@@ -24,17 +26,24 @@ requiredVars=(
 
 utils::check_required_vars "${requiredVars[@]}"
 
-# shellcheck disable=SC1090
-source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/library.sh"
-
 function cleanup() {
-  activateDefaultSa
+    # activate defauklt SA
+    client_email=$(jq -r '.client_email' < "${GOOGLE_APPLICATION_CREDENTIALS}")
+    log::info "Activating account $client_email"
+    gcloud config set account "${client_email}" || exit 1
 }
 
-shout "Authenticate as service account with write access to GCR"
-date
+log::info "Authenticate as service account with write access to GCR"
 trap cleanup EXIT
-authenticateSaGcr
+# authenticate Sa Gcr
+authKey=$1
+if [[ -n "${authKey}" ]]; then
+    client_email=$(jq -r '.client_email' < "${authKey}")
+    log::info "Authenticating in regsitry ${DOCKER_PUSH_REPOSITORY%%/*} as $client_email"
+    docker login -u _json_key --password-stdin https://"${DOCKER_PUSH_REPOSITORY%%/*}" < "${authKey}" || exit 1
+else
+    log::info "could not authenticate to Docker Registry: authKey is empty" >&2
+fi
```

Comment:
Maybe we should edit the `gcloud.sh` library with functionality of logging in to registry instead of refactoring the code we want to remove?

Reply:
I've reverted `delete-image.sh` to the original state, made new functions in `lib/gcloud.sh` (`gcloud::activate_default_sa`, `gcloud::authenticate_sa_gcr`, `gcloud::delete_docker_image`), replaced the one occurrence of `delete-image.sh` with `gcloud::delete_docker_image`.

The remaining references to `delete-image.sh` are all in `gkeCleanup` in `library.sh`, moving on to refactor this function as well